syntax = "proto3";

package specialist;

option go_package = "./pb";

service SpecialistService {
  rpc AnalyzeStream (stream SpecialistRequest) returns (SpecialistResponse);
}

message SpecialistRequest {
  oneof entry {
    Metadata metadata = 1;
    bytes chunk = 2;
  }
}

message Metadata {
    string message_id =1;
    string file_name = 2;
    string mime_type = 3;
}

// Response from the Specialist binary (m2cgen transpiled code)
message SpecialistResponse {
  oneof response {
  DocumentData document_data = 1;
  Score score = 2;
  }
}

message Score {
  double score = 1;            // Probability score between 0.0 and 1.0
  string label = 2;           // Human-readable verdict (e.g., "SPAM", "TOXIC")
}

message DocumentData {
  string title = 1;
  string author = 2;
  int32 page_count = 3;
  string language = 4;
  repeated Page pages = 5;
}

message Page {
  int32 number = 1;
  string content = 2;
}